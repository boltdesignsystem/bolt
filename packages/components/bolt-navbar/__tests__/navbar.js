import { render, stopServer } from '../../../testing/testing-helpers';
import schema from '../navbar.schema';
const { spacing, theme } = schema.properties;
let page, fixtures;

// @TODO:
// - add test for viewports, reference git history.
//   - we should be testing the hide title here.
//   - we want to be testing the show-more menu.

afterAll(async () => {
  await stopServer();
  await page.close();
}, 100);

beforeEach(async () => {
  await page.evaluate(() => {
    document.body.innerHTML = '';
  });
});

beforeAll(async () => {
  page = await global.__BROWSER__.newPage();
  await page.goto('http://127.0.0.1:4444/', {
    timeout: 0,
  });

  const itemData = [
    {
      link: {
        content: 'Section 1',
        attributes: {
          href: '#section-1',
        },
      },
    },
    {
      link: {
        content: 'Section 2',
        attributes: {
          href: '#section-2',
        },
      },
    },
    {
      link: {
        content: 'Section 3',
        attributes: {
          href: '#section-3',
        },
      },
    },
  ];

  const itemArray = [];

  await Promise.all(
    itemData.map(item =>
      render('@bolt-components-navbar/navbar-li.twig', item),
    ),
  ).then(results => {
    results.forEach(({ ok, html }) => {
      if (ok) {
        itemArray.push(html);
      }
    });
  });

  const navbarList = await render('@bolt-components-navbar/navbar-ul.twig', {
    content: itemArray.join(''),
  });

  const navbarTitle = {
    content: 'Bolt Design System',
  };

  fixtures = {
    navbarList,
    navbarTitle,
  };
});

describe('Twig Usage', () => {
  test(`Navbar default`, async () => {
    const results = await render('@bolt-components-navbar/navbar.twig', {
      title: fixtures.navbarTitle,
      links: fixtures.navbarList.html,
    });
    await expect(results.ok).toBe(true);
    await expect(results.html).toMatchSnapshot();
  });

  test(`Navbar without links`, async () => {
    const results = await render('@bolt-components-navbar/navbar.twig', {
      title: fixtures.navbarTitle,
    });
    await expect(results.ok).toBe(true);
    await expect(results.html).toMatchSnapshot();
  });

  test(`Navbar without a title`, async () => {
    const results = await render('@bolt-components-navbar/navbar.twig', {
      links: fixtures.navbarList.html,
    });
    await expect(results.ok).toBe(true);
    await expect(results.html).toMatchSnapshot();
  });
});

describe('Bolt Navbar Props', () => {
  test(`Navbar with a linked title and icon`, async () => {
    const results = await render('@bolt-components-navbar/navbar.twig', {
      title: {
        content: 'This is the navbar title',
        icon: {
          name: 'marketing-gray',
        },
        link: {
          attributes: {
            href: 'https://pega.com',
          },
        },
      },
      links: fixtures.navbarList.html,
    });
    await expect(results.ok).toBe(true);
    await expect(results.html).toMatchSnapshot();
  });

  spacing.enum.forEach(async option => {
    test(`Navbar spacing`, async () => {
      const results = await render('@bolt-components-navbar/navbar.twig', {
        title: fixtures.navbarTitle,
        links: fixtures.navbarList.html,
        spacing: option,
      });
      await expect(results.ok).toBe(true);
      await expect(results.html).toMatchSnapshot();
    });
  });

  test(`Navbar centered`, async () => {
    const results = await render('@bolt-components-navbar/navbar.twig', {
      title: fixtures.navbarTitle,
      links: fixtures.navbarList.html,
      center: true,
    });
    await expect(results.ok).toBe(true);
    await expect(results.html).toMatchSnapshot();
  });

  test(`Navbar width`, async () => {
    const results = await render('@bolt-components-navbar/navbar.twig', {
      title: fixtures.navbarTitle,
      links: fixtures.navbarList.html,
      width: 'auto',
    });
    await expect(results.ok).toBe(true);
    await expect(results.html).toMatchSnapshot();
  });

  test(`Navbar static`, async () => {
    const results = await render('@bolt-components-navbar/navbar.twig', {
      title: fixtures.navbarTitle,
      links: fixtures.navbarList.html,
      static: true,
    });
    await expect(results.ok).toBe(true);
    await expect(results.html).toMatchSnapshot();
  });

  theme.enum.forEach(async option => {
    test(`Navbar theme`, async () => {
      const results = await render('@bolt-components-navbar/navbar.twig', {
        title: fixtures.navbarTitle,
        links: fixtures.navbarList.html,
        theme: option,
      });
      await expect(results.ok).toBe(true);
      await expect(results.html).toMatchSnapshot();
    });
  });
});
