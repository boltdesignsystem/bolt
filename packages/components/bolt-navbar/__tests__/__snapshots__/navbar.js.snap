// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Bolt Navbar Props Navbar centered 1`] = `
<nav data-bolt-more-text="More"
     aria-labelledby="js-bolt-navbar-title-12345"
     class="c-bolt-navbar js-bolt-sticky c-bolt-navbar--center c-bolt-navbar--spacing-small t-bolt-dark"
>
  <div class="c-bolt-navbar__inner">
    <div class="c-bolt-navbar__title">
      <div class="c-bolt-navbar__title-inner">
        <span id="js-bolt-navbar-title-12345"
              class="c-bolt-navbar__title-text"
        >
          Bolt Design System
        </span>
      </div>
    </div>
    <div class="c-bolt-navbar__list-wrapper js-bolt-navbar__list-wrapper">
      <ul class="c-bolt-navbar__list js-bolt-navbar__list">
        <li class="c-bolt-navbar-item js-bolt-navbar-item">
          <div class="c-bolt-navbar-item__inner">
            <a href="#section-1"
               data-scroll-ignore
               class="c-bolt-navbar-item__link"
            >
              <span class="c-bolt-navbar-item__text">
                Section 1
              </span>
            </a>
          </div>
        </li>
        <li class="c-bolt-navbar-item js-bolt-navbar-item">
          <div class="c-bolt-navbar-item__inner">
            <a href="#section-2"
               data-scroll-ignore
               class="c-bolt-navbar-item__link"
            >
              <span class="c-bolt-navbar-item__text">
                Section 2
              </span>
            </a>
          </div>
        </li>
        <li class="c-bolt-navbar-item js-bolt-navbar-item">
          <div class="c-bolt-navbar-item__inner">
            <a href="#section-3"
               data-scroll-ignore
               class="c-bolt-navbar-item__link"
            >
              <span class="c-bolt-navbar-item__text">
                Section 3
              </span>
            </a>
          </div>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;

exports[`Bolt Navbar Props Navbar spacing 1`] = `
<nav data-bolt-more-text="More"
     aria-labelledby="js-bolt-navbar-title-12345"
     class="c-bolt-navbar js-bolt-sticky c-bolt-navbar--spacing-none t-bolt-dark"
>
  <div class="c-bolt-navbar__inner">
    <div class="c-bolt-navbar__title">
      <div class="c-bolt-navbar__title-inner">
        <span id="js-bolt-navbar-title-12345"
              class="c-bolt-navbar__title-text"
        >
          Bolt Design System
        </span>
      </div>
    </div>
    <div class="c-bolt-navbar__list-wrapper js-bolt-navbar__list-wrapper">
      <ul class="c-bolt-navbar__list js-bolt-navbar__list">
        <li class="c-bolt-navbar-item js-bolt-navbar-item">
          <div class="c-bolt-navbar-item__inner">
            <a href="#section-1"
               data-scroll-ignore
               class="c-bolt-navbar-item__link"
            >
              <span class="c-bolt-navbar-item__text">
                Section 1
              </span>
            </a>
          </div>
        </li>
        <li class="c-bolt-navbar-item js-bolt-navbar-item">
          <div class="c-bolt-navbar-item__inner">
            <a href="#section-2"
               data-scroll-ignore
               class="c-bolt-navbar-item__link"
            >
              <span class="c-bolt-navbar-item__text">
                Section 2
              </span>
            </a>
          </div>
        </li>
        <li class="c-bolt-navbar-item js-bolt-navbar-item">
          <div class="c-bolt-navbar-item__inner">
            <a href="#section-3"
               data-scroll-ignore
               class="c-bolt-navbar-item__link"
            >
              <span class="c-bolt-navbar-item__text">
                Section 3
              </span>
            </a>
          </div>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;

exports[`Bolt Navbar Props Navbar spacing 2`] = `
<nav data-bolt-more-text="More"
     aria-labelledby="js-bolt-navbar-title-12345"
     class="c-bolt-navbar js-bolt-sticky c-bolt-navbar--spacing-xsmall t-bolt-dark"
>
  <div class="c-bolt-navbar__inner">
    <div class="c-bolt-navbar__title">
      <div class="c-bolt-navbar__title-inner">
        <span id="js-bolt-navbar-title-12345"
              class="c-bolt-navbar__title-text"
        >
          Bolt Design System
        </span>
      </div>
    </div>
    <div class="c-bolt-navbar__list-wrapper js-bolt-navbar__list-wrapper">
      <ul class="c-bolt-navbar__list js-bolt-navbar__list">
        <li class="c-bolt-navbar-item js-bolt-navbar-item">
          <div class="c-bolt-navbar-item__inner">
            <a href="#section-1"
               data-scroll-ignore
               class="c-bolt-navbar-item__link"
            >
              <span class="c-bolt-navbar-item__text">
                Section 1
              </span>
            </a>
          </div>
        </li>
        <li class="c-bolt-navbar-item js-bolt-navbar-item">
          <div class="c-bolt-navbar-item__inner">
            <a href="#section-2"
               data-scroll-ignore
               class="c-bolt-navbar-item__link"
            >
              <span class="c-bolt-navbar-item__text">
                Section 2
              </span>
            </a>
          </div>
        </li>
        <li class="c-bolt-navbar-item js-bolt-navbar-item">
          <div class="c-bolt-navbar-item__inner">
            <a href="#section-3"
               data-scroll-ignore
               class="c-bolt-navbar-item__link"
            >
              <span class="c-bolt-navbar-item__text">
                Section 3
              </span>
            </a>
          </div>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;

exports[`Bolt Navbar Props Navbar spacing 3`] = `
<nav data-bolt-more-text="More"
     aria-labelledby="js-bolt-navbar-title-12345"
     class="c-bolt-navbar js-bolt-sticky c-bolt-navbar--spacing-small t-bolt-dark"
>
  <div class="c-bolt-navbar__inner">
    <div class="c-bolt-navbar__title">
      <div class="c-bolt-navbar__title-inner">
        <span id="js-bolt-navbar-title-12345"
              class="c-bolt-navbar__title-text"
        >
          Bolt Design System
        </span>
      </div>
    </div>
    <div class="c-bolt-navbar__list-wrapper js-bolt-navbar__list-wrapper">
      <ul class="c-bolt-navbar__list js-bolt-navbar__list">
        <li class="c-bolt-navbar-item js-bolt-navbar-item">
          <div class="c-bolt-navbar-item__inner">
            <a href="#section-1"
               data-scroll-ignore
               class="c-bolt-navbar-item__link"
            >
              <span class="c-bolt-navbar-item__text">
                Section 1
              </span>
            </a>
          </div>
        </li>
        <li class="c-bolt-navbar-item js-bolt-navbar-item">
          <div class="c-bolt-navbar-item__inner">
            <a href="#section-2"
               data-scroll-ignore
               class="c-bolt-navbar-item__link"
            >
              <span class="c-bolt-navbar-item__text">
                Section 2
              </span>
            </a>
          </div>
        </li>
        <li class="c-bolt-navbar-item js-bolt-navbar-item">
          <div class="c-bolt-navbar-item__inner">
            <a href="#section-3"
               data-scroll-ignore
               class="c-bolt-navbar-item__link"
            >
              <span class="c-bolt-navbar-item__text">
                Section 3
              </span>
            </a>
          </div>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;

exports[`Bolt Navbar Props Navbar spacing 4`] = `
<nav data-bolt-more-text="More"
     aria-labelledby="js-bolt-navbar-title-12345"
     class="c-bolt-navbar js-bolt-sticky c-bolt-navbar--spacing-medium t-bolt-dark"
>
  <div class="c-bolt-navbar__inner">
    <div class="c-bolt-navbar__title">
      <div class="c-bolt-navbar__title-inner">
        <span id="js-bolt-navbar-title-12345"
              class="c-bolt-navbar__title-text"
        >
          Bolt Design System
        </span>
      </div>
    </div>
    <div class="c-bolt-navbar__list-wrapper js-bolt-navbar__list-wrapper">
      <ul class="c-bolt-navbar__list js-bolt-navbar__list">
        <li class="c-bolt-navbar-item js-bolt-navbar-item">
          <div class="c-bolt-navbar-item__inner">
            <a href="#section-1"
               data-scroll-ignore
               class="c-bolt-navbar-item__link"
            >
              <span class="c-bolt-navbar-item__text">
                Section 1
              </span>
            </a>
          </div>
        </li>
        <li class="c-bolt-navbar-item js-bolt-navbar-item">
          <div class="c-bolt-navbar-item__inner">
            <a href="#section-2"
               data-scroll-ignore
               class="c-bolt-navbar-item__link"
            >
              <span class="c-bolt-navbar-item__text">
                Section 2
              </span>
            </a>
          </div>
        </li>
        <li class="c-bolt-navbar-item js-bolt-navbar-item">
          <div class="c-bolt-navbar-item__inner">
            <a href="#section-3"
               data-scroll-ignore
               class="c-bolt-navbar-item__link"
            >
              <span class="c-bolt-navbar-item__text">
                Section 3
              </span>
            </a>
          </div>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;

exports[`Bolt Navbar Props Navbar static 1`] = `
<nav data-bolt-static
     data-bolt-more-text="More"
     aria-labelledby="js-bolt-navbar-title-12345"
     class="c-bolt-navbar c-bolt-navbar--static c-bolt-navbar--spacing-small t-bolt-dark"
>
  <div class="c-bolt-navbar__inner">
    <div class="c-bolt-navbar__title">
      <div class="c-bolt-navbar__title-inner">
        <span id="js-bolt-navbar-title-12345"
              class="c-bolt-navbar__title-text"
        >
          Bolt Design System
        </span>
      </div>
    </div>
    <div class="c-bolt-navbar__list-wrapper js-bolt-navbar__list-wrapper">
      <ul class="c-bolt-navbar__list js-bolt-navbar__list">
        <li class="c-bolt-navbar-item js-bolt-navbar-item">
          <div class="c-bolt-navbar-item__inner">
            <a href="#section-1"
               data-scroll-ignore
               class="c-bolt-navbar-item__link"
            >
              <span class="c-bolt-navbar-item__text">
                Section 1
              </span>
            </a>
          </div>
        </li>
        <li class="c-bolt-navbar-item js-bolt-navbar-item">
          <div class="c-bolt-navbar-item__inner">
            <a href="#section-2"
               data-scroll-ignore
               class="c-bolt-navbar-item__link"
            >
              <span class="c-bolt-navbar-item__text">
                Section 2
              </span>
            </a>
          </div>
        </li>
        <li class="c-bolt-navbar-item js-bolt-navbar-item">
          <div class="c-bolt-navbar-item__inner">
            <a href="#section-3"
               data-scroll-ignore
               class="c-bolt-navbar-item__link"
            >
              <span class="c-bolt-navbar-item__text">
                Section 3
              </span>
            </a>
          </div>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;

exports[`Bolt Navbar Props Navbar theme 1`] = `
<nav data-bolt-more-text="More"
     aria-labelledby="js-bolt-navbar-title-12345"
     class="c-bolt-navbar js-bolt-sticky c-bolt-navbar--spacing-small t-bolt-xlight"
>
  <div class="c-bolt-navbar__inner">
    <div class="c-bolt-navbar__title">
      <div class="c-bolt-navbar__title-inner">
        <span id="js-bolt-navbar-title-12345"
              class="c-bolt-navbar__title-text"
        >
          Bolt Design System
        </span>
      </div>
    </div>
    <div class="c-bolt-navbar__list-wrapper js-bolt-navbar__list-wrapper">
      <ul class="c-bolt-navbar__list js-bolt-navbar__list">
        <li class="c-bolt-navbar-item js-bolt-navbar-item">
          <div class="c-bolt-navbar-item__inner">
            <a href="#section-1"
               data-scroll-ignore
               class="c-bolt-navbar-item__link"
            >
              <span class="c-bolt-navbar-item__text">
                Section 1
              </span>
            </a>
          </div>
        </li>
        <li class="c-bolt-navbar-item js-bolt-navbar-item">
          <div class="c-bolt-navbar-item__inner">
            <a href="#section-2"
               data-scroll-ignore
               class="c-bolt-navbar-item__link"
            >
              <span class="c-bolt-navbar-item__text">
                Section 2
              </span>
            </a>
          </div>
        </li>
        <li class="c-bolt-navbar-item js-bolt-navbar-item">
          <div class="c-bolt-navbar-item__inner">
            <a href="#section-3"
               data-scroll-ignore
               class="c-bolt-navbar-item__link"
            >
              <span class="c-bolt-navbar-item__text">
                Section 3
              </span>
            </a>
          </div>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;

exports[`Bolt Navbar Props Navbar theme 2`] = `
<nav data-bolt-more-text="More"
     aria-labelledby="js-bolt-navbar-title-12345"
     class="c-bolt-navbar js-bolt-sticky c-bolt-navbar--spacing-small t-bolt-light"
>
  <div class="c-bolt-navbar__inner">
    <div class="c-bolt-navbar__title">
      <div class="c-bolt-navbar__title-inner">
        <span id="js-bolt-navbar-title-12345"
              class="c-bolt-navbar__title-text"
        >
          Bolt Design System
        </span>
      </div>
    </div>
    <div class="c-bolt-navbar__list-wrapper js-bolt-navbar__list-wrapper">
      <ul class="c-bolt-navbar__list js-bolt-navbar__list">
        <li class="c-bolt-navbar-item js-bolt-navbar-item">
          <div class="c-bolt-navbar-item__inner">
            <a href="#section-1"
               data-scroll-ignore
               class="c-bolt-navbar-item__link"
            >
              <span class="c-bolt-navbar-item__text">
                Section 1
              </span>
            </a>
          </div>
        </li>
        <li class="c-bolt-navbar-item js-bolt-navbar-item">
          <div class="c-bolt-navbar-item__inner">
            <a href="#section-2"
               data-scroll-ignore
               class="c-bolt-navbar-item__link"
            >
              <span class="c-bolt-navbar-item__text">
                Section 2
              </span>
            </a>
          </div>
        </li>
        <li class="c-bolt-navbar-item js-bolt-navbar-item">
          <div class="c-bolt-navbar-item__inner">
            <a href="#section-3"
               data-scroll-ignore
               class="c-bolt-navbar-item__link"
            >
              <span class="c-bolt-navbar-item__text">
                Section 3
              </span>
            </a>
          </div>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;

exports[`Bolt Navbar Props Navbar theme 3`] = `
<nav data-bolt-more-text="More"
     aria-labelledby="js-bolt-navbar-title-12345"
     class="c-bolt-navbar js-bolt-sticky c-bolt-navbar--spacing-small t-bolt-dark"
>
  <div class="c-bolt-navbar__inner">
    <div class="c-bolt-navbar__title">
      <div class="c-bolt-navbar__title-inner">
        <span id="js-bolt-navbar-title-12345"
              class="c-bolt-navbar__title-text"
        >
          Bolt Design System
        </span>
      </div>
    </div>
    <div class="c-bolt-navbar__list-wrapper js-bolt-navbar__list-wrapper">
      <ul class="c-bolt-navbar__list js-bolt-navbar__list">
        <li class="c-bolt-navbar-item js-bolt-navbar-item">
          <div class="c-bolt-navbar-item__inner">
            <a href="#section-1"
               data-scroll-ignore
               class="c-bolt-navbar-item__link"
            >
              <span class="c-bolt-navbar-item__text">
                Section 1
              </span>
            </a>
          </div>
        </li>
        <li class="c-bolt-navbar-item js-bolt-navbar-item">
          <div class="c-bolt-navbar-item__inner">
            <a href="#section-2"
               data-scroll-ignore
               class="c-bolt-navbar-item__link"
            >
              <span class="c-bolt-navbar-item__text">
                Section 2
              </span>
            </a>
          </div>
        </li>
        <li class="c-bolt-navbar-item js-bolt-navbar-item">
          <div class="c-bolt-navbar-item__inner">
            <a href="#section-3"
               data-scroll-ignore
               class="c-bolt-navbar-item__link"
            >
              <span class="c-bolt-navbar-item__text">
                Section 3
              </span>
            </a>
          </div>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;

exports[`Bolt Navbar Props Navbar theme 4`] = `
<nav data-bolt-more-text="More"
     aria-labelledby="js-bolt-navbar-title-12345"
     class="c-bolt-navbar js-bolt-sticky c-bolt-navbar--spacing-small t-bolt-xdark"
>
  <div class="c-bolt-navbar__inner">
    <div class="c-bolt-navbar__title">
      <div class="c-bolt-navbar__title-inner">
        <span id="js-bolt-navbar-title-12345"
              class="c-bolt-navbar__title-text"
        >
          Bolt Design System
        </span>
      </div>
    </div>
    <div class="c-bolt-navbar__list-wrapper js-bolt-navbar__list-wrapper">
      <ul class="c-bolt-navbar__list js-bolt-navbar__list">
        <li class="c-bolt-navbar-item js-bolt-navbar-item">
          <div class="c-bolt-navbar-item__inner">
            <a href="#section-1"
               data-scroll-ignore
               class="c-bolt-navbar-item__link"
            >
              <span class="c-bolt-navbar-item__text">
                Section 1
              </span>
            </a>
          </div>
        </li>
        <li class="c-bolt-navbar-item js-bolt-navbar-item">
          <div class="c-bolt-navbar-item__inner">
            <a href="#section-2"
               data-scroll-ignore
               class="c-bolt-navbar-item__link"
            >
              <span class="c-bolt-navbar-item__text">
                Section 2
              </span>
            </a>
          </div>
        </li>
        <li class="c-bolt-navbar-item js-bolt-navbar-item">
          <div class="c-bolt-navbar-item__inner">
            <a href="#section-3"
               data-scroll-ignore
               class="c-bolt-navbar-item__link"
            >
              <span class="c-bolt-navbar-item__text">
                Section 3
              </span>
            </a>
          </div>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;

exports[`Bolt Navbar Props Navbar theme 5`] = `
<nav data-bolt-more-text="More"
     aria-labelledby="js-bolt-navbar-title-12345"
     class="c-bolt-navbar js-bolt-sticky c-bolt-navbar--spacing-small"
>
  <div class="c-bolt-navbar__inner">
    <div class="c-bolt-navbar__title">
      <div class="c-bolt-navbar__title-inner">
        <span id="js-bolt-navbar-title-12345"
              class="c-bolt-navbar__title-text"
        >
          Bolt Design System
        </span>
      </div>
    </div>
    <div class="c-bolt-navbar__list-wrapper js-bolt-navbar__list-wrapper">
      <ul class="c-bolt-navbar__list js-bolt-navbar__list">
        <li class="c-bolt-navbar-item js-bolt-navbar-item">
          <div class="c-bolt-navbar-item__inner">
            <a href="#section-1"
               data-scroll-ignore
               class="c-bolt-navbar-item__link"
            >
              <span class="c-bolt-navbar-item__text">
                Section 1
              </span>
            </a>
          </div>
        </li>
        <li class="c-bolt-navbar-item js-bolt-navbar-item">
          <div class="c-bolt-navbar-item__inner">
            <a href="#section-2"
               data-scroll-ignore
               class="c-bolt-navbar-item__link"
            >
              <span class="c-bolt-navbar-item__text">
                Section 2
              </span>
            </a>
          </div>
        </li>
        <li class="c-bolt-navbar-item js-bolt-navbar-item">
          <div class="c-bolt-navbar-item__inner">
            <a href="#section-3"
               data-scroll-ignore
               class="c-bolt-navbar-item__link"
            >
              <span class="c-bolt-navbar-item__text">
                Section 3
              </span>
            </a>
          </div>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;

exports[`Bolt Navbar Props Navbar width 1`] = `
<nav data-bolt-more-text="More"
     aria-labelledby="js-bolt-navbar-title-12345"
     class="c-bolt-navbar js-bolt-sticky c-bolt-navbar--auto-width c-bolt-navbar--spacing-small t-bolt-dark"
>
  <div class="c-bolt-navbar__inner">
    <div class="c-bolt-navbar__title">
      <div class="c-bolt-navbar__title-inner">
        <span id="js-bolt-navbar-title-12345"
              class="c-bolt-navbar__title-text"
        >
          Bolt Design System
        </span>
      </div>
    </div>
    <div class="c-bolt-navbar__list-wrapper js-bolt-navbar__list-wrapper">
      <ul class="c-bolt-navbar__list js-bolt-navbar__list">
        <li class="c-bolt-navbar-item js-bolt-navbar-item">
          <div class="c-bolt-navbar-item__inner">
            <a href="#section-1"
               data-scroll-ignore
               class="c-bolt-navbar-item__link"
            >
              <span class="c-bolt-navbar-item__text">
                Section 1
              </span>
            </a>
          </div>
        </li>
        <li class="c-bolt-navbar-item js-bolt-navbar-item">
          <div class="c-bolt-navbar-item__inner">
            <a href="#section-2"
               data-scroll-ignore
               class="c-bolt-navbar-item__link"
            >
              <span class="c-bolt-navbar-item__text">
                Section 2
              </span>
            </a>
          </div>
        </li>
        <li class="c-bolt-navbar-item js-bolt-navbar-item">
          <div class="c-bolt-navbar-item__inner">
            <a href="#section-3"
               data-scroll-ignore
               class="c-bolt-navbar-item__link"
            >
              <span class="c-bolt-navbar-item__text">
                Section 3
              </span>
            </a>
          </div>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;

exports[`Bolt Navbar Props Navbar with a linked title and icon 1`] = `
<nav data-bolt-more-text="More"
     aria-labelledby="js-bolt-navbar-title-12345"
     class="c-bolt-navbar js-bolt-sticky c-bolt-navbar--spacing-small t-bolt-dark"
>
  <div class="c-bolt-navbar__inner">
    <div class="c-bolt-navbar__title">
      <a href="https://pega.com"
         class="c-bolt-navbar__title-inner c-bolt-navbar__title-inner--link"
      >
        <span class="c-bolt-navbar__title-icon">
          <bolt-icon name="marketing-gray"
                     size="large"
                     aria-hidden="true"
          >
            <span class="c-bolt-icon c-bolt-icon--marketing-gray c-bolt-icon--large">
              <svg class="c-bolt-icon__icon c-bolt-icon__icon--large"
                   xmlns="http://www.w3.org/2000/svg"
                   data-name="Layer 1"
                   viewbox="0 0 32 32"
              >
                <path fill="var(--bolt-theme-icon, currentColor)"
                      fill-rule="evenodd"
                      d="M30.3 8.8H28a13.7 13.7 0 012 7.2A14 14 0 1116 2a14.5 14.5 0 017.4 2.1V1.8A15.3 15.3 0 0016 0a16 16 0 1016 16 15.6 15.6 0 00-1.7-7.2"
                      data-name="Fill-1"
                >
                </path>
                <path fill="var(--bolt-theme-icon, currentColor)"
                      fill-rule="evenodd"
                      d="M16 8a8.2 8.2 0 013.1.6l1.5-1.5A10.4 10.4 0 0016 6a10 10 0 1010 10 9.4 9.4 0 00-1.2-4.7l-1.5 1.5A8.8 8.8 0 0124 16a8 8 0 11-8-8"
                      data-name="Fill-4"
                >
                </path>
                <path fill="var(--bolt-theme-icon, currentColor)"
                      fill-rule="evenodd"
                      d="M16 16.5a.5.5 0 11.5-.5.5.5 0 01-.5.5zm.9-2.8a3 3 0 00-.9-.2 2.5 2.5 0 102.5 2.5 2.4 2.4 0 00-.2-.9L25.4 8H30V6h-2.6l1.8-1.8-1.4-1.4L26 4.5V2h-2v4.6z"
                      data-name="Fill-6"
                >
                </path>
              </svg>
            </span>
          </bolt-icon>
        </span>
        <span id="js-bolt-navbar-title-12345"
              class="c-bolt-navbar__title-text"
        >
          This is the navbar title
        </span>
      </a>
    </div>
    <div class="c-bolt-navbar__list-wrapper js-bolt-navbar__list-wrapper">
      <ul class="c-bolt-navbar__list js-bolt-navbar__list">
        <li class="c-bolt-navbar-item js-bolt-navbar-item">
          <div class="c-bolt-navbar-item__inner">
            <a href="#section-1"
               data-scroll-ignore
               class="c-bolt-navbar-item__link"
            >
              <span class="c-bolt-navbar-item__text">
                Section 1
              </span>
            </a>
          </div>
        </li>
        <li class="c-bolt-navbar-item js-bolt-navbar-item">
          <div class="c-bolt-navbar-item__inner">
            <a href="#section-2"
               data-scroll-ignore
               class="c-bolt-navbar-item__link"
            >
              <span class="c-bolt-navbar-item__text">
                Section 2
              </span>
            </a>
          </div>
        </li>
        <li class="c-bolt-navbar-item js-bolt-navbar-item">
          <div class="c-bolt-navbar-item__inner">
            <a href="#section-3"
               data-scroll-ignore
               class="c-bolt-navbar-item__link"
            >
              <span class="c-bolt-navbar-item__text">
                Section 3
              </span>
            </a>
          </div>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;

exports[`Twig Usage Navbar default 1`] = `
<nav data-bolt-more-text="More"
     aria-labelledby="js-bolt-navbar-title-12345"
     class="c-bolt-navbar js-bolt-sticky c-bolt-navbar--spacing-small t-bolt-dark"
>
  <div class="c-bolt-navbar__inner">
    <div class="c-bolt-navbar__title">
      <div class="c-bolt-navbar__title-inner">
        <span id="js-bolt-navbar-title-12345"
              class="c-bolt-navbar__title-text"
        >
          Bolt Design System
        </span>
      </div>
    </div>
    <div class="c-bolt-navbar__list-wrapper js-bolt-navbar__list-wrapper">
      <ul class="c-bolt-navbar__list js-bolt-navbar__list">
        <li class="c-bolt-navbar-item js-bolt-navbar-item">
          <div class="c-bolt-navbar-item__inner">
            <a href="#section-1"
               data-scroll-ignore
               class="c-bolt-navbar-item__link"
            >
              <span class="c-bolt-navbar-item__text">
                Section 1
              </span>
            </a>
          </div>
        </li>
        <li class="c-bolt-navbar-item js-bolt-navbar-item">
          <div class="c-bolt-navbar-item__inner">
            <a href="#section-2"
               data-scroll-ignore
               class="c-bolt-navbar-item__link"
            >
              <span class="c-bolt-navbar-item__text">
                Section 2
              </span>
            </a>
          </div>
        </li>
        <li class="c-bolt-navbar-item js-bolt-navbar-item">
          <div class="c-bolt-navbar-item__inner">
            <a href="#section-3"
               data-scroll-ignore
               class="c-bolt-navbar-item__link"
            >
              <span class="c-bolt-navbar-item__text">
                Section 3
              </span>
            </a>
          </div>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;

exports[`Twig Usage Navbar without a title 1`] = `
<nav data-bolt-more-text="More"
     class="c-bolt-navbar js-bolt-sticky c-bolt-navbar--spacing-small t-bolt-dark"
>
  <div class="c-bolt-navbar__inner">
    <div class="c-bolt-navbar__title">
      <div class="c-bolt-navbar__title-inner">
        <span class="c-bolt-navbar__title-text c-bolt-navbar__title-text--hide-from-xsmall-bp">
          Go to
        </span>
      </div>
    </div>
    <div class="c-bolt-navbar__list-wrapper js-bolt-navbar__list-wrapper">
      <ul class="c-bolt-navbar__list js-bolt-navbar__list">
        <li class="c-bolt-navbar-item js-bolt-navbar-item">
          <div class="c-bolt-navbar-item__inner">
            <a href="#section-1"
               data-scroll-ignore
               class="c-bolt-navbar-item__link"
            >
              <span class="c-bolt-navbar-item__text">
                Section 1
              </span>
            </a>
          </div>
        </li>
        <li class="c-bolt-navbar-item js-bolt-navbar-item">
          <div class="c-bolt-navbar-item__inner">
            <a href="#section-2"
               data-scroll-ignore
               class="c-bolt-navbar-item__link"
            >
              <span class="c-bolt-navbar-item__text">
                Section 2
              </span>
            </a>
          </div>
        </li>
        <li class="c-bolt-navbar-item js-bolt-navbar-item">
          <div class="c-bolt-navbar-item__inner">
            <a href="#section-3"
               data-scroll-ignore
               class="c-bolt-navbar-item__link"
            >
              <span class="c-bolt-navbar-item__text">
                Section 3
              </span>
            </a>
          </div>
        </li>
      </ul>
    </div>
  </div>
</nav>
`;

exports[`Twig Usage Navbar without links 1`] = `
<nav data-bolt-more-text="More"
     aria-labelledby="js-bolt-navbar-title-12345"
     class="c-bolt-navbar js-bolt-sticky c-bolt-navbar--spacing-small t-bolt-dark"
>
  <div class="c-bolt-navbar__inner">
    <div class="c-bolt-navbar__title">
      <div class="c-bolt-navbar__title-inner">
        <span id="js-bolt-navbar-title-12345"
              class="c-bolt-navbar__title-text"
        >
          Bolt Design System
        </span>
      </div>
    </div>
  </div>
</nav>
`;
